### PR TITLE
Factor out uploading to separate functions

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

File: lib/puppet_blacksmith/forge.rb [hunk 1]
```diff
@@ -30,13 +30,34 @@ def push!(name, package = nil)
...
       end
       raise Errno::ENOENT, "File does not exist: #{package}" unless File.exists?(package)
 
-      # login to the puppet forge
+      upload!(package)
+    end
+
+    private
+
+    def upload!(file)
```
Comment (bastelfreak) [inline lib/puppet_blacksmith/forge.rb]: why is it upload!() instead of upload()?

Comment (ekohl) [reply lib/puppet_blacksmith/forge.rb]: I guess because it has a side effect but I'm not too familiar with ruby conventions.

Comment (akofink) [reply lib/puppet_blacksmith/forge.rb]: Usually bang! methods are public - they communicate something to the user of an object's API. In this case, it's private, so you could call it dinosaur, and it's fine.

Comment (ekohl) [reply lib/puppet_blacksmith/forge.rb]: Changed to ```upload()```

**Pull request merged**